qcvm/qcvm_return.c

```diff
@@ -49,18 +52,12 @@ void qcvm_return_entity(qcvm_t *qcvm, int entity)
 /* return a string to the previous function */
 void qcvm_return_string(qcvm_t *qcvm, const char *s)
 {
-	/* bounds check */
-	if (qcvm->tempstrings_ptr + strlen(s) + 1 > qcvm->tempstrings + TEMPSTRINGS_SIZE)
-		qcvm->tempstrings_ptr = qcvm->tempstrings;
-
 	/* sprintf */
-	sprintf(qcvm->tempstrings_ptr, "%s", s);
+	memset(qcvm->tempstrings_ptr, 0, TEMPSTRINGS_SIZE);
+	memcpy(qcvm->tempstrings_ptr, s, strlen(s));
 
 	/* return str */
 	RETURN_STRING(qcvm->tempstrings_ptr);
-
-	/* advance ptr */
-	qcvm->tempstrings_ptr += strlen(s) + 1;
```

Comment:
The reason I had it set up like this, was so that functions that created a "tempstring" (i.e. strcat, vtos, ftos) could hang onto it for a while, until the buffer looped back around and it was overwritten. Probably not the greatest way to do it, maybe implementing the FTE Quake "strzone" builtin would have been better. Or growing the buffer.